src/test/java/codeu/controller/RegisterServletTest.java

```diff
@@ -13,22 +14,27 @@
 
  private RegisterServlet registerServlet;
  private HttpServletRequest mockRequest;
- private PrintWriter mockPrintWriter;
+ //private PrintWriter mockPrintWriter;
  private HttpServletResponse mockResponse;
+ private RequestDispatcher mockRequestDispatcher;
 
  @Before
  public void setup() throws IOException {
    registerServlet = new RegisterServlet();
    mockRequest = Mockito.mock(HttpServletRequest.class);
-   mockPrintWriter = Mockito.mock(PrintWriter.class);
+   //mockPrintWriter = Mockito.mock(PrintWriter.class);
    mockResponse = Mockito.mock(HttpServletResponse.class);
-   Mockito.when(mockResponse.getWriter()).thenReturn(mockPrintWriter);
+   mockRequestDispatcher = Mockito.mock(RequestDispatcher.class);
+   //Mockito.when(mockResponse.getWriter()).thenReturn(mockPrintWriter);
+   Mockito.when(mockRequest.getRequestDispatcher("/WEB-INF/view/register.jsp"))
```

Comment:
Although this is written as a stub, it works effectively as verification: it verifies that getRequestDispatcher is called with "/WEB-INF/view/register.jsp". That approach works, but the error messages are misleading because the mock just returns a null and you get a NullPointerException.

It would be better to stub this to getRequestDispatcher(any()), and then in the test call verify(mockRequest).getDispatcher("/WEB-INF/view/register.jsp"). The error message will be more informative.

(I encourage you to introduce a failure manually by changing the call to getRequestDispatcher to see what I mean.)